### ISSUE TITLE
In the docker container, Go does not work if container is ran as non-root user

### ISSUE THREAD

**Issue by maxhbr** [oss-review-toolkit/ort#4407]
The docker image specifies `$GOPATH` to `/go` which is not writable by an unprivileged user. It results in lots of `permission denied` errors.

My quickfix is to add `export GOPATH=$HOME/go` to my entrypoint.

**Comment by sschuberth** [comment]
Actually, the `Dockerfile` has

https://github.com/oss-review-toolkit/ort/blob/6978782ae9bb8e9bdc574c0fe834954c90ca8764/Dockerfile#L68

So, is `$HOME` empty in your case?

**Comment by sschuberth** [comment]
However, I just found and fixed https://github.com/oss-review-toolkit/ort/pull/4410.

**Comment by maxhbr** [comment]
I think `$HOME` is empty on build time, since when I ran the image I have a top level folder `/go` :
```
$ docker run -it --entrypoint=ls ort:latest -alF /go/bin
total 10832
drwxr-xr-x 2 root root     4096 Aug 28 16:28 ./
drwxr-xr-x 3 root root     4096 Aug 28 16:28 ../
-rwx--x--x 1 root root 11081280 Aug 28 16:28 dep*
```

But that is build-time $HOME and #4410 only has impact if the build time user changes. In my Case I later run the container with a totally different user. 

I run the docker image with 
```bash
$ docker run -i --rm \
    -v /etc/group:/etc/group:ro \
    -v /etc/passwd:/etc/passwd:ro \
    -u "$(id -u):$(id -g)" \
    -v "${HOME}/.ort/dockerHome:${HOME}" \
    -v $(pwd):/workdir \
    -w /workdir \
    --net=host ort:latest [....]
```
That allows me to write files in my host file system with correct permissions. But Go or the resolver does not work.

 But that then fails since GOPATH is already set to `/go` (with permission problems for `/go/pkgs/...`). I can fix that by setting the GOPATH explicitly to a place, where the user has permissions, but that also has disadvantages.

**Comment by heliocastro** [comment]
I did in current version of multi-docker similar thing to run as a regular user, but has the drawback that depends on one specific user id to match.
Snippet from the last part ( not commited yet though )

```
#------------------------------------------------------------------------
# Add arg user and run as a regular user
ARG USER_ID=0
ARG GROUP_ID=0

RUN if [ ${USER_ID:-0} -ne 0 ] && [ ${GROUP_ID:-0} -ne 0 ]; then \
    groupadd -g ${GROUP_ID} ort \
    && useradd -l -u ${USER_ID} -g ort ort \
    && install -d -m 0755 -o ort -g ort /home/ort; \
    fi

USER ort
WORKDIR /home/ort

```

You then need to run with same user as you set the ID. 
Is far to be optimal, but at least i can run all things as a user

**Comment by maxhbr** [comment]
Hey @heliocastro,

I think specifying the USER_ID and GROUP_ID on build time is not the right way. That should be done later and should be flexible to adopt.

**Comment by heliocastro** [comment]
Agreed, since the current multidocker is frozen, i'm working in the new
requested version and will look in a different approach

**Comment by sschuberth** [comment]
There's still a problem with `/tmp/go` not being writable when created by root at Docker image build time, but I already got a fix for that.
